android-core/src/main/kotlin/com/mparticle/internal/ApplicationContextWrapper.kt

```diff
@@ -0,0 +1,351 @@
+        registerActivityLifecycleCallbacks(callback, false)
+    }
+
+    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    fun registerActivityLifecycleCallbacks(
+        callback: ActivityLifecycleCallbacks?,
+        unitTesting: Boolean
+    ) {
+        mBaseApplication.registerActivityLifecycleCallbacks(callback)
+        val runnable = ReplayLifecycleCallbacksRunnable(callback)
+        if (unitTesting) {
+            runnable.run()
+        } else {
+            if (Looper.myLooper() == null) {
+                Looper.prepare()
+            }
+            Handler().post(runnable)
+        }
+    }
+
+    override fun getApplicationContext(): Context {
+        return this
+    }
+
+    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    override fun unregisterActivityLifecycleCallbacks(callback: ActivityLifecycleCallbacks) {
+        mBaseApplication.unregisterActivityLifecycleCallbacks(callback)
+    }
+
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+    override fun registerOnProvideAssistDataListener(callback: OnProvideAssistDataListener) {
+        mBaseApplication.registerOnProvideAssistDataListener(callback)
+    }
+
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+    override fun unregisterOnProvideAssistDataListener(callback: OnProvideAssistDataListener) {
+        mBaseApplication.unregisterOnProvideAssistDataListener(callback)
+    }
+
+    override fun hashCode(): Int {
+        return mBaseApplication.hashCode()
+    }
+
+    override fun equals(obj: Any?): Boolean {
+        return mBaseApplication == obj
+    }
+
+    override fun toString(): String {
+        return mBaseApplication.toString()
+    }
+
+    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    private fun startRecordLifecycles() {
+        stopRecordLifecycles()
+        mBaseApplication.registerActivityLifecycleCallbacks(mActivityLifecycleCallbackRecorder)
+    }
+
+    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    fun stopRecordLifecycles() {
+        mBaseApplication.unregisterActivityLifecycleCallbacks(mActivityLifecycleCallbackRecorder)
+    }
+
+    val activityLifecycleCallbackRecorderInstance: ActivityLifecycleCallbackRecorder
+        get() = ActivityLifecycleCallbackRecorder()
+
+    fun getLifeCycleEventInstance(
+        methodType: MethodType,
+        activityRef: WeakReference<Activity>?
+    ): LifeCycleEvent {
+        return LifeCycleEvent(methodType, activityRef)
+    }
+
+    fun getLifeCycleEventInstance(
+        methodType: MethodType,
+        activityRef: WeakReference<Activity>?,
+        bundle: Bundle?
+    ): LifeCycleEvent {
+        return LifeCycleEvent(methodType, activityRef, bundle)
+    }
+
+    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    inner class ActivityLifecycleCallbackRecorder : ActivityLifecycleCallbacks {
+        var lifeCycleEvents: MutableList<LifeCycleEvent> =
+            Collections.synchronizedList(LinkedList())
+        var MAX_LIST_SIZE: Int = 10
```

Comment:
```suggestion
        val MAX_LIST_SIZE: Int = 10
```
This should be a constant